arcaflow_plugin_rtla/rtla_plugin.py

```diff
@@ -0,0 +1,199 @@
+        # Index   IRQ-001   Thr-001   Usr-001   IRQ-002   Thr-002   Usr-002
+        # 0          1000       944       134       999       998       986
+        # 10            0        55       863         0         1        13
+        # 20            0         1         3         0         0         0
+        # over:         0         0         0         0         0         0
+        # count:     1000      1000      1000       999       999       999
+        # min:          0         3         4         0         2         3
+        # avg:          1         7        10         0         2         3
+        # max:          7        21        24         3        11        15
+        # ALL:        IRQ       Thr       Usr
+        # count:     1999      1999      1999
+        # min:          0         2         3
+        # avg:          0         4         6
+        # max:          7        21        24
+
+        time_unit = ""
+        col_headers = []
+        latency_hist = []
+        total_irq_latency = {}
+        total_thr_latency = {}
+        total_usr_latency = {}
+        stats_names = ["over:", "count:", "min:", "avg:", "max:"]
+        stats_per_col = []
+        found_all = False
+
+        re_isunit = re.compile(r"^#\ Time\ unit")
+        re_isindex = re.compile(r"^Index")
+        re_isdigit = re.compile(r"^\d")
+        re_isall = re.compile(r"^ALL")
+
+        for line in output.splitlines():
+            if re_isunit.match(line):
+                time_unit = line.split()[4]
+            # Capture the column headers
+            elif re_isindex.match(line):
+                col_headers = line.lower().split()
+            # Stats names repeat, so flag when have passed ^ALL
+            elif re_isall.match(line) and not found_all:
+                found_all = True
+            # Either this is a histogram bucket row, or the first time we have seen
+            # a row beginning with a stat name
+            elif (re_isdigit.match(line)) or (
+                line.split()[0] in stats_names and not found_all
+            ):
+                # Capture the columnar data
+                row_obj = {}
+                for i, col in enumerate(col_headers):
+                    row_obj[col] = line.split()[i]
+                if re_isdigit.match(line):
+                    latency_hist.append(row_obj)
+                else:
+                    stats_per_col.append(row_obj)
+            # This is the second time we have seen a row beginning with a stat name, so
+            # we want to generate key:value pairs instead of columnar data
+            elif found_all and line.split()[0] in stats_names:
+                if line.split()[0] != "over:":
+                    total_irq_latency[line.split()[0][:-1]] = line.split()[1]
+                    total_thr_latency[line.split()[0][:-1]] = line.split()[2]
+                    if params.user_threads:
+                        total_usr_latency[line.split()[0][:-1]] = line.split()[3]
```

Comment:
Consider factoring out the common subexpression, for pity on your reader if not for efficiency....  😆 
```suggestion
                label = line.split()[0][:-1]
                if label != "over":
                    total_irq_latency[label] = line.split()[1]
                    total_thr_latency[label] = line.split()[2]
                    if params.user_threads:
                        total_usr_latency[label] = line.split()[3]
```
or even
```suggestion
                items = line.split()
                label = items[0][:-1]
                if label != "over":
                    total_irq_latency[label] = items[1]
                    total_thr_latency[label] = items[2]
                    if params.user_threads:
                        total_usr_latency[label] = items[3]
```

Reply:
Another possibility...a bit arcane, but kind of fun, given that apparently today is "zip day":
```suggestion
                tables = (total_irq_latency, total_thr_latency, total_usr_latency)
...
                label = line.split()[0][:-1]
                if label != "over":
                    for table, value in zip(tables, line.split()[1:]):
                        table[label] = value
```
`zip()` will skip the usr latency if the data isn't in the line.  🙂

Reply:
The zip is cute here, but I think a bit harder to read.